spec/models/entity_spec.rb

```diff
@@ -0,0 +1,10 @@
+# spec/models/entity_spec.rb
+require 'rails_helper'
+
+RSpec.describe Entity, type: :model do
```

Comment:
- [ ] I noticed you don’t have integration test more so, your tests are few, plus they are not passing despite precompiling, I will encourage you to add more unit tests as you fix and ensure this one works before making another submission and also, add integration tests.